source/s3_util.c

```diff
@@ -488,6 +488,86 @@ int aws_s3_parse_content_length_response_header(
     return result;
 }
 
+int aws_s3_parse_request_range_header(
+    struct aws_http_headers *request_headers,
+    bool *out_initial_message_has_start_range,
+    bool *out_initial_message_has_end_range,
+    uint64_t *out_initial_start_range,
+    uint64_t *out_initial_end_range) {
+
+    AWS_PRECONDITION(request_headers);
+    AWS_PRECONDITION(out_initial_message_has_start_range);
+    AWS_PRECONDITION(out_initial_message_has_end_range);
+    AWS_PRECONDITION(out_initial_start_range);
+    AWS_PRECONDITION(out_initial_end_range);
+
+    bool initial_message_has_start_range = false, initial_message_has_end_range = false;
+    uint64_t initial_start_range = 0, initial_end_range = 0;
```

Comment:
I don't think it's written in our coding standards, but never initialize more than one variable per line. It's very error-prone (people forget to initialize one of the earlier variables)
```suggestion
    bool initial_message_has_start_range = false;
    bool initial_message_has_end_range = false;
    uint64_t initial_start_range = 0;
    uint64_t initial_end_range = 0;
```

Reply:
Thanks, updated.